### ISSUE TITLE
Working around the file lock

### ISSUE THREAD

**Issue by bitingsock** [yt-dlp/yt-dlp#3124]
### Checklist

- [X] I'm reporting a feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

prior to commit https://github.com/yt-dlp/yt-dlp/commit/747c0bd127ebd205278d31ec7216ffa02fe96734
it was possible to play a file while it was being downloaded. Is there a known workaround to return the behavior?

**Comment by pukkandan** [comment]
The way I understand it, locked file should still be playable. But I'll double check

**Comment by bitingsock** [comment]
for reference I only tried mpv and vlc, both ffmpeg players

**Comment by pukkandan** [comment]
You are using windows, correct?

**Comment by bitingsock** [comment]
Yes. compiled binary version

**Comment by bitingsock** [comment]
I don't know for sure it was that commit that changed the behavior.
possibly: https://github.com/yt-dlp/yt-dlp/commit/a3125791c7a5cdf2c8c025b99788bf686edd1a8a

**Comment by pukkandan** [comment]
> possibly: a312579

Yes, that is the right commit. I have found the issue, but still trying to figure out a solution. There may actually be no proper solution other than adding a --no-lock-files option

**Comment by jakeogh** [comment]
Are Win* locks not advisory? Either way, `--no-lock-files` would be nice to have for testing, I'll send a pull.

**Comment by pukkandan** [comment]
For now, I have disabled locking on windows completely (except for archives). PRs for adding `--(no-)lock-file` options are welcome

**Comment by Vangelis66** [comment]
> I have disabled locking on windows completely (except for archives).

Thanks ! :+1: 

Disabling `locked_file`, first introduced, as correctly reported, in a312579, should restore the `--hls-use-mpegts` switch, too... 

I only yesterday night discovered, much to my dismay, that the partially downloading .ts file (HLSe stream) couldn't be played/accessed by **MPC-HC**; I searched the issue tracker for "`--hls-use-mpegts` broken" but couldn't find any related issue (by title) ... 

Some hours ago I decided to dissect the breakage by sequentially trying the "daily" Windows builds kindly offered by **Lesmiscore**, after a (long) while I arrived to the **last good** build, `v2022.02.05.114514`; after examining the changelog between that and the next (bad) build, `v2022.02.06.114514`, 

https://github.com/ytdl-patched/yt-dlp/compare/2022.02.05.114514...2022.02.06.114514

I concluded myself that the culprit was:

https://github.com/ytdl-patched/yt-dlp/commit/a3125791c7a5cdf2c8c025b99788bf686edd1a8a

FWIW, I never myself use `yt-dlp` in the way that necessitated #1066, but I understand if others want to... 

RL issues and, at the same time, other digital tasks of a different nature don't allow me to follow as closely as I would've liked/hoped the development of this fine app 😭 ; but my best wishes and immense gratitude go to all the team members 👍 😄

**Comment by pukkandan** [comment]
Yes, the behavior of locking on windows was an oversight on my part. I knew that windows only has mandatory locking, but (wrongly) assumed that there is a way to block only write access and not reads. Technically, I was right and that is possible with non-exclusive lock; but then, even the handle that is locking the file cannot write to it, making it useless for us.
